src/main/groovy/org/ods/doc/gen/project/data/ProjectData.groovy

```diff
@@ -190,13 +191,16 @@ class ProjectData {
     }
 
     private Map<String, String> loadMetadataRepo(repo) {
+        if ((! repo) || (! repo.id)) {
+            throw new RuntimeException("Repository id cannot be blank or null.")
+        }
         return  [
                 id: repo.id,
-                name: repo.name,
-                description: "myDescription-A",
-                supplier: "mySupplier-A",
-                version: "myVersion-A",
-                references: "myReferences-A"
+                name: repo.name ? repo.name : repo.id,
```

Comment:
In the other class you said repo.name:
 String repoName = repo.id as String
            repoName = "${projectId}-${repoName}"
Here is different?
And when is possible to don't have the repo.name?

Reply:
This needs more work. 
Seems to be in the original commit you forgot about this information. 
We are now determining how to obtain this information.
Thanks for pointing to it.